webview.h

```diff
@@ -911,6 +939,25 @@ class cocoa_wkwebview_engine {
     objc_registerClassPair(cls);
     return objc::msg_send<id>((id)cls, "new"_sel);
   }
+  id create_webkit_navigation_delegate() {
+    auto cls = objc_allocateClassPair((Class) "NSObject"_cls,
+                                      "WebkitNavigationDelegate", 0);
+    class_addProtocol(cls, objc_getProtocol("WKNavigationDelegate"));
+    class_addMethod(cls, "webView:didFinishNavigation:"_sel,
+                    (IMP)(+[](id delegate, SEL sel, id webview, id navigation) {
+                      auto w = get_associated_webview(delegate);
```

Comment:
We need to be careful here with memory management given the potential frequency of the execution of this code. What does the memory footprint look like?

Reply:
good question, not too familiar with the apple profiling frameworks.

Reply:
Something like this should give an idea about the memory consumption:

* Record the app's memory usage after a single navigation.
* Repeat navigation a bunch of times, whether with a local server (e.g. `python3 -m http.server --bind 127.0.0.1 8000`) or any other way that triggers the code to execute, and watch memory consumption. Not sure if it's enough to call `navigate()` or if we need to automate mouse/keyboard taps.
* After the loop, see whether memory consumption goes down after a few seconds. It's pretty lazy.
* If memory consumption goes down to normal levels then it should be fine.

If this looks OK then I would just skip profiling because we probably still have some memory-related issues on macOS that aren't too severe.